src/promptflow/promptflow/azure/_storage/cosmosdb/summary.py

```diff
@@ -87,14 +90,39 @@ def persist(self, client):
             SpanAttributeFieldName.REFERENCED_BATCH_RUN_ID in attributes
             and SpanAttributeFieldName.LINE_NUMBER in attributes
         ):
-            # Add sleep to wait for main run to be persisted.
-            # We receive requests to persist main flow first and then evaluation,
-            # but init cosmosDB client is time consuming, and the later request will reuse the same client, so it's
-            # possible that the main run is not persisted when we start to patch evaluation to it.
-            time.sleep(1)
-            self._insert_evaluation(client)
-
-    def _persist_line_run(self, client):
+            self._insert_evaluation_with_retry(client)
+
+    # When there is the first span for line run, write placeholder item to LineSummary table.
+    def _persist_running_item(self, client: ContainerProxy):
+        trace_id = self.span.trace_id
+        session_id = self.span.session_id
+
+        # TODO: Refer same Running const with SDK
+        item = SummaryLine(
+            id=trace_id,
+            partition_key=session_id,
+            session_id=session_id,
+            trace_id=trace_id,
+            status="Running",
+            created_by=self.created_by,
+        )
+        attributes: dict = self.span._content[SpanFieldName.ATTRIBUTES]
+        if SpanAttributeFieldName.LINE_RUN_ID in attributes:
+            item.line_run_id = attributes[SpanAttributeFieldName.LINE_RUN_ID]
+        elif SpanAttributeFieldName.BATCH_RUN_ID in attributes and SpanAttributeFieldName.LINE_NUMBER in attributes:
+            item.batch_run_id = attributes[SpanAttributeFieldName.BATCH_RUN_ID]
+            item.line_number = attributes[SpanAttributeFieldName.LINE_NUMBER]
+        try:
```

Comment:
do we need to take care eval test/batch line here? for those traces with attrs `referenced.xxx_run_id`